src/main/java/org/hmhb/config/ConfigController.java

```diff
@@ -0,0 +1,90 @@
+package org.hmhb.config;
+
+import javax.annotation.Nonnull;
+import javax.servlet.http.HttpServletResponse;
+
+import com.codahale.metrics.annotation.Timed;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Spring MVC {@link Controller} for serving our our common configuration
+ * between the client and server in a javascript file (so the client can
+ * load this synchronously and use the data in angular's config phase).
+ */
+@RestController
+public class ConfigController {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ConfigController.class);
+
+    private final HttpServletResponse response;
+    private final ConfigService configService;
+    private final ObjectMapper objectMapper;
+
+    /**
+     * An injectable constructor.
+     *
+     * @param response TODO
+     * @param configService the {@link ConfigService} for config
+     * @param objectMapper the {@link ObjectMapper} to convert our config to
+     *                     JSON for the client
+     */
+    @Autowired
+    public ConfigController(
+            @Nonnull HttpServletResponse response,
+            @Nonnull ConfigService configService,
+            @Nonnull ObjectMapper objectMapper
+    ) {
+        this.response = requireNonNull(response, "response cannot be null");
+        this.configService = requireNonNull(configService, "configService cannot be null");
+        this.objectMapper = requireNonNull(objectMapper, "objectMapper cannot be null");
+    }
+
+    /**
+     * Serves our single page app for any sub-views.
+     *
+     * @return our single page app's index.html
+     * @throws JsonProcessingException if the {@link ObjectMapper} failed to
+     * serialize the config
+     */
+    @Timed
+    @RequestMapping(
+            method = RequestMethod.GET,
+            value = "/config.js"
+    )
```

Comment:
Will this be returned as content type "application/javascript"? I can't remember if Spring does this for you with *.js